### PR TITLE
#9 add devise

### PR DESCRIPTION
**Pull request by MattGomb**
Dear Code Reviewer,

hi-gif

This PR is about [Blog app: add devise] (https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_devise.md).

In this PR we followed the below instructions:
- Remove current_user method in ApplicationController, devise will provide us with one.
- [Install and setup devise](https://github.com/heartcombo/devise#getting-started).
- Can register a new user.
- User logs in with a combination of email and password.
- Hashed password should be stored in the database.
- Ask for confirmation of email.
- Can reset password.
[Modify the views in devise](https://github.com/heartcombo/devise#configuring-views) for registration and login to match [the wireframes from the Sneak Peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md#:~:text=For%20this%20project%20you%20will%20have%20full%20freedom%20in%20terms%20of%20visual%20design%20but%20you%20will%20need%20to%20keep%20the%20following%20wireframes%3A) and your styling.

### CODE REVIEW

File: app/views/devise/registrations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      <%= f.password_field :password_confirmation, autocomplete: "new-password", placeholder: 'Confirm Password' %>
+    </div>
+
+    <div class="actions">
+      <%= f.submit "Sign up", class: 'signup-btn btn' %>
+      <% end %>
+
+      <%= render "devise/shared/links" %>
+    </div>
+</div>
```
Comment (GzimAsani) [inline app/views/devise/registrations/new.html.erb]: - [x] Check the image for more info 
image

Comment (GzimAsani) [reply app/views/devise/registrations/new.html.erb]: - [x] Check the image for more info 
image

File: app/views/devise/sessions/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<%= render 'shared/flashes' %>
+<div class="container-b">
```
Comment (GzimAsani) [inline app/views/devise/sessions/new.html.erb:2]: - [x] Kindly follow the given wire frame for user index page

Wireframe:
image
Yours:
image

Comment (GzimAsani) [reply app/views/devise/sessions/new.html.erb:2]: - [x] For user show page please follow the given wireframe
image
image

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello, @MattGomb            :wave:   
 Good job so far,  done There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with adding devise :+1: 

 ## Required Changes 

- [ ] Check the comments below :arrow_double_down: 

 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/devise/registrations/new.html.erb (hunk 1 above)
Comment (MattGomb) [reply app/views/devise/registrations/new.html.erb]: Styling fixed

File: app/views/devise/sessions/new.html.erb (hunk 2 above)
Comment (MattGomb) [reply app/views/devise/sessions/new.html.erb:2]: fixed

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## There is nothing else to say than it's time to merge... :partying_face: 
Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


## Optional suggestions


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by MattGomb** [comment]
Thanks for the quick reviews!

**Pull request merged**